File: P Olimpico de Programação (G4M3) IFPB/sortudosounao.py
n, x = map(int, input().split())
l = list(map(int, input().split()))

esquerda = 0
direita = 0
soma = 0

ok = False
while direita < n:
  soma += l[direita]
  direita+=1
  while soma > x and esquerda < direita:
    soma -= l[esquerda]
    esquerda += 1
  if soma == x:
    ok = True
    break

if ok:
  print("YES")
else: 
  print("NO")

File: P Olimpico de Programação (G4M3) IFPB/sortudosounao.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, x; cin >> n >> x;
    int esquerda = 0, direita = 0, soma = 0; bool ok = false;
    vector<int> l(n);
    for (int i = 0; i < n; i++) {
        cin >> l[i];
    }
    while (direita < n) {
        soma += l[direita];
        direita++;
        while (soma > x && esquerda < direita) {
            soma -= l[esquerda];
            esquerda++;
        }
        if (soma == x) {
            ok = true;
            break;
        }
    }
    if (ok) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }
    return 0;
}
